File: 2014/nazionali/bufale/att/grader.c
#include <stdio.h>
#include <assert.h>
#define MAXN 10000000

static int N, i;
static int M[MAXN], P[MAXN];

long long solve(int N, int* M, int* P);

static FILE *fin, *fout;

int main(){
    fin = fopen("input.txt", "r");
    fout = fopen("output.txt", "w");
    assert(1 == fscanf(fin, "%d", &N));
    for(i=0; i<N; i++) assert(2 == fscanf(fin, "%d%d", M+i, P+i));
    fprintf(fout, "%lld\n", solve(N, M, P));
    fclose(fin);
    fclose(fout);
    return 0;
}

File: 2014/nazionali/bufale/att/grader.pas
uses bufale;

const MAXN = 10000000;
var N, i      : longint;
    M, P      : array[0..MAXN-1] of longint;
    fin, fout : text;

begin
    assign(fin, 'input.txt'); reset(fin);
    assign(fout, 'output.txt'); rewrite(fout);
    read(fin, N);
    for i:=0 to N-1 do read(fin, M[i], P[i]);
    writeln(fout, solve(N, M, P));
    close(fin);
    close(fout);
end.
